File: FinalProject/app/src/main/java/com/example/finalproject/Potion.java
package com.example.finalproject;

public class Potion extends Item {
    public int lasting_time, strength, effect;

    public Potion(int effect, String name) {
        this.effect = effect;
        super.setName(name);
    }

    public Potion(int lasting_time, int strength, int effect) {
        this.lasting_time = lasting_time;
        this.strength = strength;
        this.effect = effect;
    }

    public int getLasting_time() {
        return lasting_time;
    }

    public void setLasting_time(int lasting_time) {
        this.lasting_time = lasting_time;
    }

    public int getStrength() {
        return strength;
    }

    public void setStrength(int strength) {
        this.strength = strength;
    }

    public int getEffect() {
        return effect;
    }

    public void setEffect(int effect) {
        this.effect = effect;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/service/classes/items/Potion.kt
package com.example.finalproject.service.classes.items

class Potion(
    name: String,
    id: Int,
    costSell: Int,
    costBuy: Int,
    rarity: Int,
    category: Int,
    val lastingTime: Int,
    val strength: Double,
    val effect: Int
) :
    Item(name, id, costSell, costBuy, rarity, category)